src/cart/crt_hg_proc.c

```diff
@@ -781,12 +800,16 @@ crt_proc_out_common(crt_proc_t proc, crt_rpc_output_t *data)
 	/* D_DEBUG("in crt_proc_out_common, data: %p\n", *data); */
 
 	if (proc_op != CRT_PROC_FREE) {
+		if (proc_op == CRT_PROC_ENCODE)
+			rpc_priv->crp_reply_hdr.cch_hlc = crt_hlc_get();
```

Comment:
So this is recording when the RPC was replied to, and putting in it the header?  Would it be better to call crt_hlc_get() in crt_reply_send() and just store the previous value here?

Reply:
This function will be called right before sending reply into the wire (When Mercury will encode the packet). So, I think this is right place.

Reply:
This is the kind of information that should be in the header file for the public API, that the RPC send timestamp is the point in time that the RPC was transmitted from the local node.

Reply:
done